asset/src/s3_reader_api/api.ts

```diff
@@ -1,12 +1,14 @@
 import {
     APIFactory, AnyObject, isNil, isString, getTypeOf
 } from '@terascope/job-components';
-import S3Reader from './s3-api';
+import { S3Reader } from '@terascope/file-asset-apis';
 import { S3ReaderAPIConfig } from './interfaces';
 
 export default class S3ReaderAPI extends APIFactory<S3Reader, S3ReaderAPIConfig> {
     validateConfig(input: AnyObject): S3ReaderAPIConfig {
         if (isNil(input.path) || !isString(input.path)) throw new Error(`Invalid parameter path: it must be of type string, was given ${getTypeOf(input.path)}`);
+        // add in default here
+        if (input.compression !== 'none') input.file_per_slice = true;
```

Comment:
Are sure we want to enforce, `file_by_slice` everytime compression is not equal to none

Reply:
this was the original behaviour https://github.com/terascope/file-assets/blob/v0.14.3/asset/src/s3_reader/slicer.ts#L26

Reply:
Still seems odd to never give a choice though?